sky/backends/cloud_vm_ray_backend.py

```diff
@@ -380,10 +343,57 @@ def add_gang_scheduling_placement_group(
                 print('INFO: All task resources reserved.',
                       file=sys.stderr,
                       flush=True)
-                job_lib.set_job_started({self.job_id!r})
                 """)
         ]
 
+        job_id = self.job_id
+        if setup_cmd is not None:
+            self._code += [
+                textwrap.dedent(f"""\
+                setup_cmd = {setup_cmd!r}
+                _SETUP_CPUS = 0.0001
+                # The setup command will be run as a ray task with num_cpus=_SETUP_CPUS as the
+                # requirement; this means Ray will set CUDA_VISIBLE_DEVICES to an empty string.
+                # We unset it so that user setup command may properly use this env var.
+                setup_cmd = 'unset CUDA_VISIBLE_DEVICES; ' + setup_cmd
+                job_lib.set_status({job_id!r}, job_lib.JobStatus.SETTING_UP)
+                job_lib.scheduler.schedule_step()
```

Comment:
We don't need to call `schedule_step()` here right? Should we call it after the `ray.get(pg.ready())` in L342?

Also, do we need to call `async_schedule_step` instead?